### PR TITLE
fix: add header offset overflow check

### PR DESCRIPTION
**Pull request by zamazan4ik**
- during the header offset calculation, perform overflow check

Tested:
- Local tests

### CODE REVIEW

**Comment by zamazan4ik** [comment]
A little later I want to add additional test case for the issue.

File: tests/issue_234.rs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+#[test]
+fn invalid_date() {
+    let reader = std::io::Cursor::new(&BUF);
+    let archive = zip::ZipArchive::new(reader);
+    match archive {
+        Err(ZipError::InvalidArchive(_)) => {}
+        _ => {
+            panic!(
+                "Unexpected result: expected {:?}",
```
Comment (5225225) [inline tests/issue_234.rs:31]: Might be worth printing what the result you *did* get was?

If you're debugging this, you'd end up changing the code to include that anyways.

Comment (zamazan4ik) [reply tests/issue_234.rs:31]: Good point. I will fix it in the next commit. Thanks!

Comment (zamazan4ik) [reply tests/issue_234.rs]: Fixed

**Pull request merged**